src/main/java/cart/controller/ProductController.java

```diff
@@ -0,0 +1,58 @@
+package cart.controller;
+
+import cart.controller.dto.ProductEditRequest;
+import cart.controller.dto.ProductRequest;
+import cart.controller.dto.ProductResponse;
+import cart.controller.dto.ProductsResponse;
+import cart.domain.Product;
+import cart.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@RequiredArgsConstructor
+@Validated
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    @GetMapping
+    public ResponseEntity<ProductsResponse> getProducts() {
+        List<Product> products = productService.findAll();
+        ProductsResponse response = ProductsResponse.of(products.stream().map(product -> ProductResponse.of(product)).collect(Collectors.toList()));
```

Comment:
map 부분에서 레퍼런스 함수로 호출이 가능합니다 :)

```suggestion
        ProductsResponse response = ProductsResponse.of(products.stream().map(ProductResponse::of).collect(Collectors.toList()));
```